CabalHelper/Compile.hs

```diff
@@ -413,6 +413,9 @@ errorInstallCabal cabalVer _distdir = panic $ printf "\
  where
    sver = showVersion cabalVer
 
+helperPrepared :: Version -> IO Bool
+helperPrepared compCabalVersion = isJust <$> cachedExe compCabalVersion
```

Comment:
>  but ghc-mod caches cabal-helper info and the cache only gets invalidated when the cabal file changes

Well, we can (and, probably, should do) the same on haskell-ide-egnine side, so this is an orthogonal problem.

So, where are we now? Do you want to accept this? If yes, what should be done to get this in? Turn `compileHelper` into `IO (FilePath, IO (Maybe ExitCode))`?